libs/librepcb/libraryeditor/lib/libraryoverviewwidget.cpp

```diff
@@ -254,6 +268,64 @@ void LibraryOverviewWidget::updateElementList(QListWidget& listWidget,
   }
 }
 
+void LibraryOverviewWidget::openContextMenuAtPos(const QPoint& pos) noexcept {
+  // Build the context menu
+  QMenu    menu;
+  QAction* aRemove = menu.addAction(QIcon(":/img/actions/delete.png"), tr("Remove"));
+
+  // Execute the context menu
+  QAction* action = menu.exec(QCursor::pos());
+
+  // Get selected item
+  QListWidget* list = dynamic_cast<QListWidget*>(sender());
+  Q_ASSERT(list);
+  QListWidgetItem* selectedItem = list->itemAt(pos);
+
+  // Get item text
+  const QString itemName = selectedItem->text();
+
+  // Find file path of selected library element
+  FilePath itemPath = selectedItem ? FilePath(selectedItem->data(Qt::UserRole).toString()) : FilePath();
+  if (!itemPath.isValid()) {
+      qWarning() << "File path for selected item is not valid";
+      return;
+  }
+
+  // Handle action
+  if (action == aRemove) {
+    if (removeSelectedItem(itemName, itemPath)) {
+      delete selectedItem;
+    };
+  }
+}
+
+bool LibraryOverviewWidget::removeSelectedItem(const QString& itemName, const FilePath& itemPath) noexcept {
+  int ret = QMessageBox::warning(
+      this, tr("Remove %1").arg(itemName),
+      QString(tr("WARNING: Library elements must normally NOT be removed "
+                 "because this will break "
+                 "other elements which depend on this one! They should be just "
+                 "marked as "
+                 "deprecated instead.\n\nAre you still sure to delete the "
+                 "whole library element "
+                 "\"%1\"?\n\nThis cannot be undone!"))
+          .arg(itemName),
+      QMessageBox::Yes, QMessageBox::Cancel);
+  if (ret == QMessageBox::Yes) {
+    try {
+      FileUtils::removeDirRecursively(itemPath);
```

Comment:
If the selected element is currently opened in the library editor (as a tab), it must be closed before removing its directory.